src/Compilers/CSharp/Test/Syntax/Parsing/ScriptParsingTests.cs

```diff
@@ -9236,38 +9236,34 @@ public void GlobalStatementSeparators_Comma2()
 a < b,
 void goo() { }
 ",
-                // (2,6): error CS1002: ; expected
-                // a < b,
-                Diagnostic(ErrorCode.ERR_SemicolonExpected, ",").WithLocation(2, 6),
-                // (2,6): error CS7017: Member definition, statement, or end-of-file expected
-                // a < b,
-                Diagnostic(ErrorCode.ERR_GlobalDefinitionOrStatementExpected, ",").WithLocation(2, 6));
+                // (3,1): error CS1547: Keyword 'void' cannot be used in this context
+                // void goo() { }
+                Diagnostic(ErrorCode.ERR_NoVoidHere, "void").WithLocation(3, 1),
+                // (3,6): error CS1003: Syntax error, '>' expected
+                // void goo() { }
+                Diagnostic(ErrorCode.ERR_SyntaxError, "goo").WithArguments(">").WithLocation(3, 6));
+
             N(SyntaxKind.CompilationUnit);
```

Comment:
this change doesn't seem great.  but hte code is pretty messed up to begin with.  so maybe it's ok.

Reply:
The heuristic here would involve detecting the newline as a possible non-type argument scenario. I opted out of complicating error recovery for this purpose, but would probably be nice to consider as a scenario in the future.